pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3167,7 +3188,7 @@ public synchronized void triggerCompaction()
                 currentCompaction = brokerService.pulsar().getStrategicCompactor()
                         .compact(topic, strategicCompactionMap.get(topic));
             } else {
-                currentCompaction = brokerService.pulsar().getCompactor().compact(topic);
+                currentCompaction = topicCompactionService.compact().thenApply(x -> null);
```

Comment:
Why did you add the `thenApply` here? I think `compact()` always completes with `null`.

Reply:
Because the currentCompaction type is `CompletableFuture<Long>`, however topicCompactionService.compact() return `CompletableFuture<Void>` type.